src/minidump.rs

```diff
@@ -1513,6 +1557,200 @@ impl MinidumpAssertion {
+            .or(Err(Error::StreamReadFailure))?;
+
+        let value = match raw.ty {
+            md::MINIDUMP_ANNOTATION::TYPE_INVALID => MinidumpAnnotation::Invalid,
+            md::MINIDUMP_ANNOTATION::TYPE_STRING => {
+                let string = read_string_utf8(&mut (raw.value as usize), all, endian)
+                    .or(Err(Error::StreamReadFailure))?
+                    .to_owned();
+
+                MinidumpAnnotation::String(string)
+            }
+            _ if raw.ty >= md::MINIDUMP_ANNOTATION::TYPE_USER_DEFINED => {
+                MinidumpAnnotation::UserDefined(raw)
+            }
+            _ => MinidumpAnnotation::Unsupported(raw),
+        };
+
+        dictionary.insert(key.to_owned(), value);
+    }
+
+    Ok(dictionary)
+}
+
+impl MinidumpModuleCrashpadInfo {
+    pub fn read(
+        link: md::MINIDUMP_MODULE_CRASHPAD_INFO_LINK,
+        all: &[u8],
+        endian: scroll::Endian,
+    ) -> Result<Self, Error> {
+        let raw: md::MINIDUMP_MODULE_CRASHPAD_INFO = all
+            .pread_with(link.location.rva as usize, endian)
+            .or(Err(Error::StreamReadFailure))?;
+
+        let list_annotations = read_string_list(all, &raw.list_annotations, endian)?;
+        let simple_annotations =
+            read_simple_string_dictionary(all, &raw.simple_annotations, endian)?;
+        let annotation_objects = read_annotation_objects(all, &raw.annotation_objects, endian)?;
+
+        Ok(Self {
+            raw,
+            module_index: link.minidump_module_list_index as usize,
+            list_annotations,
+            simple_annotations,
+            annotation_objects,
+        })
+    }
+}
+
+fn read_crashpad_module_links(
+    all: &[u8],
+    location: &md::MINIDUMP_LOCATION_DESCRIPTOR,
+    endian: scroll::Endian,
+) -> Result<Vec<MinidumpModuleCrashpadInfo>, Error> {
+    let data = location_slice(all, location).or(Err(Error::StreamReadFailure))?;
+    if data.is_empty() {
+        return Ok(Vec::new());
+    }
+
+    let mut offset = 0;
+
+    let count: u32 = data
+        .gread_with(&mut offset, endian)
+        .or(Err(Error::StreamReadFailure))?;
+
+    let mut module_links = Vec::with_capacity(count as usize);
+    for _ in 0..count {
+        let link: md::MINIDUMP_MODULE_CRASHPAD_INFO_LINK = data
+            .gread_with(&mut offset, endian)
+            .or(Err(Error::StreamReadFailure))?;
+
+        let info = MinidumpModuleCrashpadInfo::read(link, all, endian)?;
+        module_links.push(info);
+    }
+
+    Ok(module_links)
+}
+
+impl<'a> MinidumpStream<'a> for MinidumpCrashpadInfo {
+    const STREAM_TYPE: MINIDUMP_STREAM_TYPE = MINIDUMP_STREAM_TYPE::CrashpadInfoStream;
+
+    fn read(bytes: &'a [u8], all: &'a [u8], endian: scroll::Endian) -> Result<Self, Error> {
+        let raw: md::MINIDUMP_CRASHPAD_INFO = bytes
+            .pread_with(0, endian)
+            .or(Err(Error::StreamReadFailure))?;
+
+        if raw.version > md::MINIDUMP_CRASHPAD_INFO::VERSION {
```

Comment:
This seems incorrect per [the Crashpad docs](https://crashpad.chromium.org/doxygen/structcrashpad_1_1MinidumpCrashpadInfo.html#a481042efad46164dc43d9c824df9e6cf):
> Upon encountering a value greater than kVersion, a reader should assume that the structure’s layout is compatible with the structure defined as having value kVersion.

We have something similar already in the handling of the versioned [`MINIDUMP_MISC_INFO_`](https://github.com/luser/rust-minidump/blob/0759d861d60374379388d587d59b1df645a8b056/src/minidump.rs#L1192) structs.